frontend/components/commons/forms/questions-form/QuestionsForm.component.vue

```diff
@@ -251,121 +242,103 @@ export default {
-        RESPONSE_STATUS_FOR_API.SUBMITTED
-      );
-
-      // 3 - Create the formatted requests to send from the formattedSelectionObject
-      const formattedRequestsToSend = this.formatRequestsToSend(
-        createOrUpdateResponse,
-        formattedSelectionObject
-      );
-
-      // 4 - create or update the record responses and emit bus event to change record to show
-      await this.createOrUpdateResponsesAndEmitRecordToGoBusEvent(
-        RECORD_STATUS.SUBMITTED,
-        formattedRequestsToSend
-      );
-    },
-    initFlagCreateOrUpdateResponse() {
-      const createOrUpdateResponse = isResponsesByUserIdExists(
-        this.userId,
-        this.recordId
-      )
-        ? STATUS_RESPONSE.UPDATE
-        : STATUS_RESPONSE.CREATE;
-
-      return createOrUpdateResponse;
-    },
-    async createOrUpdateResponsesAndEmitRecordToGoBusEvent(
-      status,
-      requestsToSend
-    ) {
       try {
-        await this.createOrUpdateRecordResponses(requestsToSend);
+        const responseValues = this.factoryInputsToResponseValues();
 
-        // NOTE - Update dataset Metrics orm
-        await this.initMetricsAndInsertDataInOrm();
+        await this.sendBackendRequest({
+          status: RESPONSE_STATUS_FOR_API.SUBMITTED,
+          values: responseValues,
+        });
 
-        // NOTE - onSubmit event => the status change to SUBMITTED
-        await updateRecordStatusByRecordId(this.recordId, status);
+        await this.refreshMetrics();
 
-        let typeOfEvent = null;
-        switch (status) {
-          case RECORD_STATUS.SUBMITTED:
-            typeOfEvent = TYPE_OF_EVENT.ON_SUBMIT;
-            break;
-          case RECORD_STATUS.DISCARDED:
-            typeOfEvent = TYPE_OF_EVENT.ON_DISCARD;
-            break;
-          default:
-            console.log(`The event ${status} is unknown`);
-        }
+        await updateRecordStatusByRecordId(
+          this.recordId,
+          RECORD_STATUS.SUBMITTED
+        );
 
-        this.onEmitBusEventGoToRecordIndex(typeOfEvent);
-      } catch (err) {
-        console.log(err);
+        this.emitBusEventGoToRecordIndex();
+      } catch (error) {
+        console.log(error);
       }
     },
-
-    onEmitBusEventGoToRecordIndex(typeOfEvent) {
-      switch (typeOfEvent) {
-        case TYPE_OF_EVENT.ON_SUBMIT:
-        case TYPE_OF_EVENT.ON_DISCARD:
-          this.$root.$emit("go-to-record-index", this.recordIdIndex + 1);
-          break;
-        default:
-      }
+    emitBusEventGoToRecordIndex() {
+      this.$root.$emit("go-to-record-index", this.recordIdIndex + 1);
     },
     async onClear() {
-      const responseId = getRecordResponsesIdByRecordId({
-        userId: this.userId,
-        recordId: this.recordId,
-      });
-
       try {
         const responseData =
-          responseId && (await this.deleteResponsesByResponseId(responseId));
+          this.responseId && (await this.deleteResponsesByResponseId());
```

Comment:
Again, if you have a method called `deleteResponsesByResponseId` you should pass the responseId as 

```suggestion
          this.responseId && (await this.deleteResponsesByResponseId(this.responseId));
```